elide-async/src/main/java/com/yahoo/elide/async/operation/TableExportCallableOperation.java

```diff
@@ -6,51 +6,193 @@
+import com.yahoo.elide.core.exceptions.TransactionException;
+import com.yahoo.elide.core.request.EntityProjection;
 import com.yahoo.elide.core.security.User;
 
 import io.reactivex.Observable;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Date;
+import java.util.UUID;
 
 import java.util.concurrent.Callable;
 
 /**
  * TableExport Execute Operation Interface.
  */
+@Slf4j
 public abstract class TableExportCallableOperation implements Callable<AsyncAPIResult> {
+    private TableExportFormatter formatter;
+    @Getter private AsyncExecutorService service;
+    private Integer recordNumber = 0;
+    private TableExport exportObj;
+    private RequestScope scope;
+
+    public TableExportCallableOperation (TableExportFormatter formatter, AsyncExecutorService service,
+            AsyncAPI exportObj, RequestScope scope) {
+        this.formatter = formatter;
+        this.service = service;
+        this.exportObj = (TableExport) exportObj;
+        this.scope = scope;
+    }
+
+    @Override
+    public AsyncAPIResult call() {
+        String apiVersion = scope.getApiVersion();
+        log.debug("TableExport Object from request: {}", exportObj);
+
+        EntityProjection projection = getProjection(exportObj, apiVersion);
+
+        Observable<PersistentResource> observableResults = export(exportObj, scope, projection);
+
+        Observable<String> results = Observable.empty();
+        String preResult = formatter.preFormat(projection, exportObj);
+        results = observableResults.map(resource -> {
+            this.recordNumber++;
+            return formatter.format(resource, recordNumber);
+        });
+        String postResult = formatter.postFormat(projection, exportObj);
+
+        // Stitch together Pre-Formatted, Formatted, Post-Formatted results of Formatter in single observable.
+        Observable<String> finalResults = Observable.empty();
+
+        if (preResult != null) {
+            finalResults = Observable.just(preResult).concatWith(results);
+        } else {
+            finalResults = results;
+        }
+
+        if (postResult != null) {
+            finalResults.concatWith(Observable.just(postResult));
+        }
+
+        storeResults(exportObj, service.getResultStorageEngine(), results);
+
+        URL downloadURL;
+        try {
+            downloadURL = new URL(generateDownloadURL(exportObj, (RequestScope) scope));
+        } catch (MalformedURLException e) {
+            log.debug("Generating download url failure. {}", e.getMessage());
+            throw new IllegalStateException(e);
+        }
+
+        TableExportResult queryResult = new TableExportResult();
+        queryResult.setHttpStatus(200);
+        queryResult.setCompletedOn(new Date());
+        queryResult.setUrl(downloadURL);
+        queryResult.setRecordCount(recordNumber);
```

Comment:
This record count does not include the Header. Should we include it?

Reply:
No.  I don't think so.  But isn't the query result set in some other base class?

Reply:
TableExportResult and AsyncQueryResult have different properties in the query result so the operation classes (TableExportCallableOperation and AsyncQueryCallableOperation) set them separately.